src/components/Snapshots/SnapshotsListView/SnapshotsColumnManagement.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import ColumnManagement from '../../../shared/components/table/ColumnManagement';
+import { snapshotColumns, defaultVisibleColumns } from './SnapshotsListHeader';
+import { SnapshotColumnKey } from './types';
+
+const NON_HIDABLE_COLUMNS: SnapshotColumnKey[] = ['name', 'kebab'];
+
+type SnapshotsColumnManagementProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  visibleColumns: Set<SnapshotColumnKey>;
+  onVisibleColumnsChange: (columns: Set<SnapshotColumnKey>) => void;
+};
+
+const SnapshotsColumnManagement: React.FC<SnapshotsColumnManagementProps> = ({
+  isOpen,
+  onClose,
+  visibleColumns,
+  onVisibleColumnsChange,
+}) => {
+  return (
+    <ColumnManagement
+      isOpen={isOpen}
+      onClose={onClose}
+      visibleColumns={visibleColumns}
+      onVisibleColumnsChange={onVisibleColumnsChange}
+      columns={snapshotColumns}
+      defaultVisibleColumns={defaultVisibleColumns}
+      nonHidableColumns={NON_HIDABLE_COLUMNS}
+      title="Manage columns"
```

Comment:
since both `title` and `description` props have default values exactly as the texts you're using here, you could simply not set here :)